cli-config/src/config.rs

```diff
@@ -12,13 +22,35 @@ lazy_static! {
     };
 }
 
+/// The Solana CLI configuration.
 #[derive(Serialize, Deserialize, Debug, PartialEq)]
 pub struct Config {
+    /// The RPC address of a Solana validator node.
+    ///
+    /// Typical values for mainnet, devnet, and testnet are [described in the
+    /// Solana documentation][rpcdocs].
+    ///
+    /// For local testing, the typical value is `http://localhost:8899`.
+    ///
+    /// [rpcdocs]: https://docs.solana.com/cluster/rpc-endpoints
     pub json_rpc_url: String,
+    /// The address to connect to for receiving event notifications.
+    ///
+    /// If it is an empty string then the correct value will be derived
+    /// from `json_rpc_url`.
+    ///
+    /// The default value is the empty string.
     pub websocket_url: String,
+    /// The path to the keypair file.
```

Comment:
```suggestion
    /// The path to the default signer.
```
Do you think it would be helpful to link to the DefaultSigner struct here? For some context about what this signer is used for.

Reply:
Actually, do you think the change I proposed is confusing? Even though we are still using the field `keypair_path`, it can take a signer path of another type, like `usb://ledger`

Probably we should deprecate the field `keypair_path` and update to `signer_path`. I think it might be too disruptive to change the CLI argument `-k`/`--keypair`, though.

Reply:
Ah this is all good info I wasn't aware of. I've been using read_keypair_file, as in my example, to load this as a Keypair, but I guess I should actually be using ... DefaultSigner::new to load this as any kind of Signer? Is that right? That's going to get a lot more complex to explain sadly, but I'll figure it out.

I don't think I can speak to changing the API as I don't know what the policies are, but it does seem beyond this one name, there's quite a bit of legacy cruft in the Solana APIs that would be nice to break at some point. For this case, we might be able to just define a completely different struct, and direct serde to make the signer_path deserializer work with the old keypair_path field. I don't think we should do it in this patch though.

Reply:
The signer_from_path free function would also work as an example, and maybe that would be a more general example? Not every app needs to care about the DefaultSigner details. What do you think?

Reply:
I can mention them both of course.

One problem I have is I don't think I can hyperlink to any of these keypair/signer loading functions from this crate, just mention them by name.

Reply:
I'll figure out some way to link to DefaultSigner etc, though it may not link to exactly the correct version.

Reply:
> I don't think we should do it in this patch though

Oh, agreed, this patch can be purely docs. I was mostly thinking out loud about how that should be improved.

`signer_from_path` seems pretty good, although now that I'm pondering it in context with these docs, I don't love how that method requires ArgMatches. It might be nice to have a utility function that does all the various signer generation without a dependency on clap.

I agree that not every app needs to care about the DefaultSigner details.

> One problem I have is I don't think I can hyperlink to any of these keypair/signer loading functions from this crate, just mention them by name.

Np, mention by name works as well, if it's too clunky to link wrt versions.